include/boost/uuid/uuid_io.hpp

```diff
@@ -146,23 +146,38 @@ inline wchar_t to_wchar(size_t i) {
 
 } // namespace detail
 
-inline std::string to_string(uuid const& u)
+template<class OutputIterator>
+OutputIterator to_chars(uuid const& u, OutputIterator out)
 {
-    std::string result;
-    result.reserve(36);
-
     std::size_t i=0;
     for (uuid::const_iterator it_data = u.begin(); it_data!=u.end(); ++it_data, ++i) {
         const size_t hi = ((*it_data) >> 4) & 0x0F;
-        result += detail::to_char(hi);
+        *out++ = detail::to_char(hi);
 
         const size_t lo = (*it_data) & 0x0F;
-        result += detail::to_char(lo);
+        *out++ = detail::to_char(lo);
 
         if (i == 3 || i == 5 || i == 7 || i == 9) {
-            result += '-';
+            *out++ = '-';
         }
     }
+    return out;
+}
+
+inline bool to_chars(uuid const& u, char* first, char* last)
```

Comment:
I'm not a huge fan of return value checking for success... should this perhaps throw something instead like a std::length_error should the buffer size be insufficient?

Reply:
I think a return value is fine for a low level tool like this.

Reply:
This was intended to be similar to the API of https://en.cppreference.com/w/cpp/utility/to_chars (non-throwing and non-allocating) but I am open to changing it.